feature-wallet-impl/src/main/java/jp/co/soramitsu/feature_wallet_impl/presentation/balance/assetActions/buy/BuyMixinProvider.kt

```diff
@@ -6,42 +6,48 @@ import jp.co.soramitsu.feature_wallet_api.domain.model.BuyTokenRegistry
 import jp.co.soramitsu.feature_wallet_impl.presentation.balance.assetActions.buy.BuyMixin.IntegrationPayload
 import jp.co.soramitsu.feature_wallet_impl.presentation.balance.assetActions.buy.BuyMixin.Presentation
 import jp.co.soramitsu.feature_wallet_impl.presentation.balance.assetActions.buy.BuyMixin.ProviderChooserPayload
+import jp.co.soramitsu.runtime.multiNetwork.ChainRegistry
 import jp.co.soramitsu.runtime.multiNetwork.chain.model.Chain
 import jp.co.soramitsu.runtime.multiNetwork.chain.model.ChainId
 
-// TODO wallet - buy
 class BuyMixinProvider(
     private val buyTokenRegistry: BuyTokenRegistry,
+    private val chainRegistry: ChainRegistry
 ) : Presentation {
 
     override val showProviderChooserEvent = MutableLiveData<Event<ProviderChooserPayload>>()
 
     override val integrateWithBuyProviderEvent = MutableLiveData<Event<IntegrationPayload>>()
 
-    override fun isBuyEnabled(chainId: ChainId, chainAssetId: Int): Boolean {
-        return false
-    }
+    override fun isBuyEnabled(chainId: ChainId, chainAssetId: Int) =
+        when (val asset = chainRegistry.getAsset(chainId, chainAssetId)) {
+            null -> false
+            else -> buyTokenRegistry.availableProviders(asset).isNotEmpty()
+        }
```

Comment:
Maybe  use

`chainRegistry.getAsset(chainId, chainAssetId)?.let { buyTokenRegistry.availableProviders(asset).isNotEmpty() 
} ?: false`

as an analog to reduce the row count, but don't know what would be better and more readable. @posplaw what do you think?